src/dynarec/rv64/dynarec_rv64_66.c

```diff
@@ -702,6 +702,26 @@ uintptr_t dynarec64_66(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                     UFLAG_RES(ed);
                     UFLAG_DF(x3, d_shr16);
                     break;
+                case 4:
+                case 6:
+                    if(opcode==0xD1) {
+                        INST_NAME("SHL Ew, 1");
+                        MOV32w(x4, 1);
+                    } else {
+                        INST_NAME("SHL Ew, CL");
+                        ANDI(x4, xRCX, 0x1f);
+                    }
+                    UFLAG_IF {MESSAGE(LOG_DUMP, "Need Optimization for flags\n");}
+                    SETFLAGS(X_ALL, SF_PENDING);
+                    GETEW(x1, 0);
+                    UFLAG_OP12(ed, x4)
+                    SLL(ed, ed, x4);
```

Comment:
you can use option `rv64_zbb` path here, with `zexth(...)`

Reply:
indeed

Reply:
Done, I searched and replaced all `SLLI + SRLI` pairs.

Reply:
Oh, wait, you have to make it optionnal. Zbb is not always supported. Use the `if(rv64_zbb)` Or change the `ZEXTH(...)` opcode itself to check for zbb and use SLLi/SRLI as a fallback.